### ISSUE TITLE
nx-patch v5.2 BETA testing

### ISSUE THREAD

**Issue by vasile-gh** [ottokiksmaler/nx500_nx1_modding#59]
@arspr @Hirsti @RieGo here's the [link](https://github.com/ottokiksmaler/nx500_nx1_modding/tree/dev/video-bitrate-mods/nx-patch) to the beta.
Can you try it and report issues in this thread?
Thanks a lot
Vasile

**Comment by hirsti** [comment]
I have tested on NX1.
Install works ok, no need to pull battery after install, menu can be invoked first time with double LCD light.
First option on menu, resolutions doesn't work, when you select it the menu disappears and the resolutions menu does not appear.
New bitrate menu options work ok, telnet works, reports ip if connected to WiFi and doesn't if not.
Focus Stack runs ok but doesn't give a focused stacking message when running and auto focus doesn't work after it has finished.

**Comment by RieGo** [comment]
same results here installation worked fine without battery pull. telnet works.
resolution menu doesn't work.
bitrate slots nq2 and nq3 worked fine.
with nq1 i had a weird problem: tested with 1 MBit/s it results in ~2 testing with 2 results in ~3 testing with 10 results in 10. maybe there is some kind of minimum bitrate?

if you activate power save to 30 seconds and play around in the mod menu camera will turn off after 30 seconds no matter what you do. if you then wake up the cam you end up with an invisible mod menu (which can be operated as normal)

**Comment by vasile-gh** [comment]
@ottokiksmaler is it worth me "fixing" the focus issue after stacking or will you release soon the new focus stacking I think I saw you saying you work on?

**Comment by ottokiksmaler** [comment]
So, if should record the focus settings before switching to MF and then restore them? 
Regarding distance scale - I have been unable to force it to update - it always lags. Sorry

**Comment by vasile-gh** [comment]
@ottokiksmaler yes, if possible. IMO it is better to try to have separate programs restore camera to "before" state. Unless you disagree :-)

**Comment by vasile-gh** [comment]
@hirsti @RieGo @arspr new beta posted. resolutions should work now - sorry that was due to the new bitrates structure - I also changed by accident a part related to resolutions. BTW the new bitrates are like that because for NX500 there were not enough slots (and @ottokiksmaler I know making more slots would make buttons too small :-) )

**Comment by RieGo** [comment]
yep, resolutions work fine now

**Comment by ottokiksmaler** [comment]
I agree. Will do it.

**Comment by hirsti** [comment]
Yes, install works ok and resolutions work ok.
Everything else is as stated above.
fast turn around vasile, nice work :-)

**Comment by hirsti** [comment]
I have found a problem if you redirect VGA to DC and choose 640x480 MJPEG.
When you start the recording the live footage is very laggy and the counter stops counting up but is still indicating it is recording.  You cannot stop the recording, pressing the record button does not stop it and even switching off the camera switch doesn't effect, you have to pull the battery which results in a corrupt file.  Other VGA entries work ok but not the MJPEG one.

**Comment by vasile-gh** [comment]
@hirsti yeah MJPEG is dealt with through another mechanism. I am actually surprised it did not crash on you.

**Comment by vasile-gh** [comment]
@ottokiksmaler should I wait on you for the new focus stacking or should I release "as is"?

**Comment by vasile-gh** [comment]
@hirsti @RieGo @arspr can you please list your cameras and let me know on which ones you want to run the beta. I'd like to get to 2-3xNX1 and 2-3xNX500 not more.

**Comment by RieGo** [comment]
i only own a NX1.

**Comment by ottokiksmaler** [comment]
As focus stacking is meant to be used in manual anyway, you can release it as it is. The other issue is that my car is at service and my laptop is in trunk so I can't do anything right now :(

**Comment by hirsti** [comment]
I only own the NX1

**Comment by arspr** [comment]
I'm on NX1 too. (No NX500).
I feel you are on the short side on NX500 testers, although I think that ottokiksmaler can count by 3 or 4 all by himself...

I've just installed 5.2 beta (I suppose 2nd version). Perfect access to the conflictive menu too.

About the focus stack "issues". I would release it as it is just now. The most serious issue (the final silent MF) has an extremely easy workaround and fix, basically change AF mode in the way you want (a AF->MF->AF cycle in the lens, re select your current SAF/CAF/AAF...). Therefore, I think it only needs a mention in the readme file till it's finally "fixed" (if it is ever fixed...)

The other two "issues", (focus in progress message not showing up and the lag in the focus distance indicator), are _extremely_ minor.

Nevertheless, a consideration I also made to Kino Seed, that I do really think you should add to the README file. **Ottokiksmaler's focus "tricks" are capable of changing focus distance _on a lens set in MF_**. More over, ottokiksmaler has just said that's the intended way for them to be used... For you, coders, maybe that's natural but from a pure photographer point of view **it's PRETTY WEIRD**. I mean, if my lens is in MF, I do not expect that the camera will be able to move the focus distance it has. Period.

**Comment by ottokiksmaler** [comment]
Sorry, didn't know about mf switch being overridden, I don't have such a lens. What I meant to say was that usually focus stacking is done in mf (but I was thinking in camera mf like with pz or 30mm lens). 
Side note - so mf switch is just a signal to camera and in no way avidly disables AF.

**Comment by arspr** [comment]
@ottokiksmaler, no need to apologize in any way. There's no problem with the way your mods work. It adds functionality to the camera so in fact **IT'S GOOD**. I only say it needs clarification because I cannot expect my lens moving the focus distance on its own if it's on MF. So I won't even think that I can set Near/Far positions in MF and then let the camera do its work unless being explicitly told. I mean, even if I set the Near/Far positions in MF, I would have probably switched to AF _before_ starting the focus stack which is a useless step (well, that's what I did before knowing it...)

Short story: moving the focus distance in MF is good but the photographer must be told it can happen...
